docs: `@pankod/refine-cli` (#3097)
refinedev/refine: documentation/docs/packages/documentation/cli.md

---
id: cli
title: CLI
---

refine CLI is a command line application that allows you to interact with your **refine** project and perform some tasks. This includes adding a new resource, running runners (build, start, dev), managing updates and swizzle components.

## Installation

Install the [@pankod/refine-cli](https://github.com/refinedev/refine/tree/master/packages/cli) library.

```bash
npm i @pankod/refine-cli
```

<Tabs
    defaultValue="cra"
    values={[
        {label: 'Create React App', value: 'cra'},

>         {label: 'Nextjs', value: 'nextjs'},

Reviewer: ```suggestion
        {label: 'Next.js', value: 'nextjs'},
```

<TabItem value="nextjs">

```diff title="package.json"
{
    "scripts": {
-        "dev": "next dev",
-        "build": "next build",
-        "start": "next start",
+        "dev": "refine dev",
+        "build": "refine build",
+        "start": "refine start",
+        "refine": "refine"
    }
}    
```
</TabItem>
<TabItem value="remix">

```diff title="package.json"
{
    "scripts": {
-       "dev": "remix dev",
-       "build": "remix build",
-       "start": "remix-serve build"
+       "dev": "refine dev",
+       "build": "refine build",
+       "start": "refine start",
+       "refine": "refine"
    }
}    
```
</TabItem>
</Tabs>

## Commands
### swizzle
### create-resource

Use this command to add a new resource to your project.

```bash
npm run refine create-resource [resourceName]
```

| Argument     | Description                               |
| ------------ | ----------------------------------------- |
| resourceName | The name of the resource you want to add. |

#### Options

| Alias | Option    | Description                                                           |
| ----- | --------- | --------------------------------------------------------------------- |

> | -a    | --actions | Only generate the specified actions. (default: list,create,edit,show) |
> | -p    | --path    | Path to generate the resource files (default: "src/pages")            |
> | -h    | --help    | Output useage information                                             |

Reviewer: ```suggestion
| -a    | --actions | Only generate the specified actions. (default: `list`,`create`,`edit`,`show`) |
| -p    | --path    | Path to generate the resource files (default: `src/pages`)            |
| -h    | --help    | Output usage information                                             |
```

| resourceName (optinal) | The name of the resource you want to add. |

#### Options

| Alias | Option    | Default                                                            | Description                                                                                |
| ----- | --------- | ------------------------------------------------------------------ | ------------------------------------------------------------------------------------------ |
| -a    | --actions | `list`,`create`,`edit`,`show`                                      | Only generate the specified actions.                                                       |
| -p    | --path    | cra: `src/pages` next.js: `src/components` remix: `app/components` | The path to create source files. (It is created automatically according to the framework.) |
| -h    | --help    |                                                                    | Output usage information                                                                   |
 
#### Examples

Let's create a `Category` resource with all the actions.

```bash
npm run refine create-resource category
```

> The following files are produced.

Reviewer 2: ```suggestion
The following files are generated.
```

                ├────────────────────────────────┼─────────┼────────┼────────┤
                │ @pankod/refine-cli             │ 1.5.1   │ 1.5.3  │ 1.5.3  │
                ├────────────────────────────────┼─────────┼────────┼────────┤
                │ @pankod/refine-core            │ 3.88.1  │ 3.90.4 │ 3.90.4 │
                ├────────────────────────────────┼─────────┼────────┼────────┤
                │ @pankod/refine-react-hook-form │ 3.31.0  │ 3.33.2 │ 3.33.2 │
                ├────────────────────────────────┼─────────┼────────┼────────┤
                │ @pankod/refine-simple-rest     │ 2.6.0   │ 2.7.8  │ 3.35.2 │
                ├────────────────────────────────┼─────────┼────────┼────────┤
                │ @pankod/refine-strapi          │ 3.18.0  │ 3.37.0 │ 3.37.0 │
                └────────────────────────────────┴─────────┴────────┴────────┘
                    To update `refine` packages with the wanted version
                        Run the following command npx refine update
```

### dev
Starts the development server of your `react` app. It will detect the framework you are using and start the development server accordingly.

> If you are using  `remix`, It will start `remix build` command.

Reviewer 2: ```suggestion
For example, if you are using `remix`, It will start the `remix dev` command.
```

```bash
npm run refine dev
```

### build
Create a production build of your `react` app. It will detect the framework you are using and build the app accordingly. 

> If you are using  `remix`, It will start `remix build` command.

Reviewer 2: For example, if you are using `remix`, It will start the `remix build` command.

```bash
npm run refine build
```

### start
Starts the production server of your `react` app. It will detect the framework you are using and start the production server accordingly. 

> If you are using  `create-react-app`, It will start `react-scripts start` command.

Reviewer 2: ```suggestion
For example, if you are using  `create-react-app`, It will start `react-scripts start` command.
```

                │ name                           │ current │ wanted │ latest │
                ├────────────────────────────────┼─────────┼────────┼────────┤
                │ @pankod/refine-airtable        │ 2.1.1   │ 2.7.8  │ 3.33.0 │
                ├────────────────────────────────┼─────────┼────────┼────────┤
                │ @pankod/refine-cli             │ 1.5.1   │ 1.5.3  │ 1.5.3  │
                ├────────────────────────────────┼─────────┼────────┼────────┤
                │ @pankod/refine-core            │ 3.88.1  │ 3.90.4 │ 3.90.4 │
                ├────────────────────────────────┼─────────┼────────┼────────┤
                │ @pankod/refine-react-hook-form │ 3.31.0  │ 3.33.2 │ 3.33.2 │
                ├────────────────────────────────┼─────────┼────────┼────────┤
                │ @pankod/refine-simple-rest     │ 2.6.0   │ 2.7.8  │ 3.35.2 │
                ├────────────────────────────────┼─────────┼────────┼────────┤
                │ @pankod/refine-strapi          │ 3.18.0  │ 3.37.0 │ 3.37.0 │
                └────────────────────────────────┴─────────┴────────┴────────┘
                    To update `refine` packages with the wanted version

>                         Run the following command npx refine update

Reviewer: ```suggestion
                        Run the following command npm run refine update
```

```

### dev,  start,  build
When you run `npm run refine [dev | start | build]` It will detect the framework you are using and run the commands accordingly.

> Also you can pass environment variables, and all the options that are available in the framework. For example, you can run `NODE_ENV=production npm run refine dev --port 3001` to run the app on port 3001.

Reviewer: ```suggestion
Also you can pass environment variables, and all the options that are available in the framework. For example, you can run `NODE_ENV=production npm run refine dev --port 3001` to run the app on port `3001`.
```

<Tabs
    defaultValue="cra"
    values={[

>         {label: 'Create React App', value: 'cra'},

Reviewer: ```suggestion
        {label: 'React', value: 'cra'},
```

        {label: 'Next.js', value: 'nextjs'},
        {label: 'Remix', value: 'remix'}
    ]}
>
<TabItem value="cra"> 

> [Refer to the Create React App documentation for detailed usage. &#8594](https://create-react-app.dev/docs/available-scripts)

Reviewer: ```suggestion
[Refer to the Create React App documentation for detailed usage. &#8594](https://create-react-app.dev/docs/available-scripts)

[Refer to the Vite documentation for detailed usage. &#8594](https://vitejs.dev/guide/#command-line-interface)
```

```bash

>  # Starts application in development mode. Equivalent to `react-scripts start`.

Reviewer: ```suggestion
 # Starts application in development mode. Equivalent to `react-scripts start` or `vite`.
```

npm run refine dev
```

```bash

> # Creates a production build of your app. Equivalent to `react-scripts build`.

Reviewer: ```suggestion
# Creates a production build of your app. Equivalent to `react-scripts build` or `vite build`.
```

---
id: cli
title: CLI
---

refine CLI is a command line application that allows you to interact with your **refine** project and perform some tasks. This includes adding a new resource, running runners (build, start, dev), managing updates and swizzle components.

## Installation

Install the [@pankod/refine-cli](https://github.com/refinedev/refine/tree/master/packages/cli) library.
It is available by default if you create a project with `npm create refine-app`.
We also recommend adding it as a dependency.

> **How to add for existing project?**

Reviewer: ```suggestion
**How to add to an existing project?**
```
Can we make an accordion?

https://docusaurus.io/docs/next/markdown-features/react#markdown-and-jsx-interoperability
